addons/io/org.openhab.io.hueemulation/src/main/java/org/openhab/io/hueemulation/internal/dto/HueDevice.java

```diff
@@ -352,10 +352,35 @@ private void setState(State itemState) {
             successApplied.put("effect", newState.effect); // Pretend that worked
         }
         if (newState.xy != null) {
-            errorApplied.add("xy");
+            try {
+                HueStateColorBulb c = as(HueStateColorBulb.class);
+                c.colormode = ColorMode.xy;
+                c.bri = as(HueStateBulb.class).bri;
+                c.xy[0] = newState.xy.get(0);
+                c.xy[1] = newState.xy.get(1);
+                command = c.toHSBType();
+                successApplied.put("xy", newState.xy);
+            } catch (ClassCastException e) {
+                errorApplied.add("xy");
+            }
         }
         if (newState.xy_inc != null) {
-            errorApplied.add("xy_inc");
+            try {
+                HueStateColorBulb c = as(HueStateColorBulb.class);
+                double newX = c.xy[0] + newState.xy_inc.get(0);
+                double newY = c.xy[1] + newState.xy_inc.get(1);
+                if (newX < 0 || newX > 1 || newY < 0 || newY > 1) {
+                    throw new ClassCastException();
+                }
+                c.colormode = ColorMode.xy;
+                c.bri = as(HueStateBulb.class).bri;
+                c.xy[0] = newX;
+                c.xy[1] = newY;
+                command = c.toHSBType();
+                successApplied.put("xy", newState.xy_inc);
+            } catch (ClassCastException e) {
```

Comment:
Can we just do an `instanceof` instead of catching this?

Reply:
That was actually copied from the lines above, look at the full sourcecode to see how it's done there.  @davidgraeff Is there any reason for this? I think it is because the exception is caught later on the return the error response, but I am not sure

Reply:
The problem in this method is, that the current device state could be a Plug, a White Bulb or a Coloured Bulb. In a first iteration I had so many `instanceof`s that I have chosen to go with the current code instead.